public/app/features/teams/TeamSettings.tsx

```diff
@@ -50,40 +50,53 @@ export const TeamSettings = ({ team, updateTeam }: Props) => {
-          </Field>
-
-          {contextSrv.licensedAccessControlEnabled() && canListRoles && (
-            <Field label={t('teams.team-settings.label-role', 'Role')}>
-              <TeamRolePicker teamId={team.id} roleOptions={roleOptions} disabled={!canUpdateRoles} maxWidth="100%" />
+          <Stack direction="column" gap={2}>
+            <Field
+              noMargin
+              label={t('teams.team-settings.label-numerical-identifier', 'Numerical identifier')}
+              disabled={true}
+            >
+              <Input value={team.id} id="id-input" />
+            </Field>
+            <Field
+              noMargin
+              label={t('teams.team-settings.label-name', 'Name')}
+              disabled={!canWriteTeamSettings || !!team.isProvisioned}
+              required
+              invalid={!!errors.name}
+              error="Name is required"
+            >
+              <Input {...register('name', { required: true })} id="name-input" />
             </Field>
-          )}
 
-          <Field
-            label={t('teams.team-settings.label-email', 'Email')}
-            description={t(
-              'teams.team-settings.description-email',
-              'This is optional and is primarily used to set the team profile avatar (via gravatar service)'
+            {contextSrv.licensedAccessControlEnabled() && canListRoles && (
+              <Field noMargin label={t('teams.team-settings.label-role', 'Role')}>
+                <TeamRolePicker teamId={team.id} roleOptions={roleOptions} disabled={!canUpdateRoles} maxWidth="100%" />
+              </Field>
             )}
-            disabled={!canWriteTeamSettings}
-          >
-            {/* eslint-disable-next-line @grafana/i18n/no-untranslated-strings */}
-            <Input {...register('email')} placeholder="team@email.com" type="email" id="email-input" />
-          </Field>
-          <Button type="submit" disabled={!canWriteTeamSettings}>
-            <Trans i18nKey="teams.team-settings.save">Save</Trans>
-          </Button>
+
+            <Field
+              noMargin
+              label={t('teams.team-settings.label-email', 'Email')}
+              description={t(
+                'teams.team-settings.description-email',
+                'This is optional and is primarily used to set the team profile avatar (via gravatar service)'
+              )}
+              disabled={!canWriteTeamSettings}
+            >
+              <Input
+                {...register('email')}
+                // eslint-disable-next-line @grafana/i18n/no-untranslated-strings
+                placeholder="team@email.com"
+                type="email"
+                id="email-input"
+              />
+            </Field>
+          </Stack>
         </FieldSet>
+        <Button type="submit" disabled={!canWriteTeamSettings}>
+          <Trans i18nKey="teams.team-settings.save">Save team details</Trans>
```

Comment:
Renamed the save button so it's distinguished from the `Save preferences` button that's sometimes shown at the same time (from a11y standpoint, it's clearer to label these two more clearly)